### PR TITLE
fixed flaky api tests

### PR DESCRIPTION
**Pull request by glower**
**What this PR does / why we need it**:

After the serialization to the json string the order is not guaranteed, so instead of comparing json strings we should compare objects 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by glower** [comment]
I have removed all direct json string compare from the tests to avoid problems with the random order after serialization/deserialization

File: api/pkg/handler/user_test.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func TestGetUsersForProject(t *testing.T) {
 			if res.Code != tc.HTTPStatus {
 				t.Fatalf("Expected HTTP status code %d, got %d: %s", tc.HTTPStatus, res.Code, res.Body.String())
 			}
-			compareWithResult(t, res, tc.ExpectedResponse)
+			compareJSON(t, res, tc.ExpectedResponse)
```
Comment (p0lyn0mial) [inline api/pkg/handler/user_test.go]: it helps ? I think that the lister `userLister.List` uses a map as its underlying store. In go iterating over a map is not stable, it's random.

Comment (p0lyn0mial) [reply api/pkg/handler/user_test.go]: similar to https://github.com/kubermatic/kubermatic/blob/master/api/pkg/handler/project_test.go#L123

Comment (glower) [reply api/pkg/handler/user_test.go]: In `compareJSON` we use `Semantic.DeepEqual` and I thought it ignores the order of the array, but it does not. We can compare maps with it but not arrays.  In this case (if we have an array of structs) we need to sort the array let say buy id. If this is OK I will implement this

Comment (p0lyn0mial) [reply api/pkg/handler/user_test.go]: yeah, we can unmarshal the response to an array of object. Then we can compare it against an expected one. I think we have more than one way of doing this. We could use a double loop or sort the lists.

Comment (glower) [reply api/pkg/handler/user_test.go]: > yeah, we can unmarshal the response to an array of object. Then we can compare it against an expected one.

This is already done in this function. I will just sort the array if it is one

**Pull request closed without merging**